src/vm/wren_core.c

```diff
@@ -472,6 +482,11 @@ DEF_PRIMITIVE(fiber_yield1)
 
 DEF_PRIMITIVE(fn_new)
 {
+  RETURN_ERROR("Fn metaclass does not implement 'new()'.");
+}
```

Comment:
one possible solution could be to create those class completely in C (by doing what `classDefinition()` does).